src/dpl/src/Opendp.cpp

```diff
@@ -377,6 +377,58 @@ void Opendp::groupInitPixels2()
   }
 }
 
+dbInst* Opendp::getAdjacentInstance(dbInst* inst, bool left) const
+{
+  const Rect core = grid_->getCore();
+  const Rect inst_rect = inst->getBBox()->getBox();
+  GridX left_x = grid_->gridX(DbuX{inst_rect.xMin() - 1 - core.xMin()});
+  GridX right_x = grid_->gridX(DbuX{inst_rect.xMax() + 1 - core.xMin()});
+
+  GridY y = grid_->gridSnapDownY(DbuY{inst_rect.yMin() - core.yMin()});
+
+  Pixel* pixel
+      = left ? grid_->gridPixel(left_x, y) : grid_->gridPixel(right_x, y);
+
+  dbInst* adjacent_inst = nullptr;
+
+  if (pixel != nullptr) {
+    // do not return macros, endcaps and tapcells
+    if (pixel->cell && pixel->cell->db_inst_->isCore()) {
```

Comment:
https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/docs/contrib/CodingPractices.md#practice-20